javagen/src/main/java/com/azure/autorest/util/MethodUtil.java

```diff
@@ -55,4 +70,155 @@ public static boolean isMethodIncludeRepeatableRequestHeaders(ProxyMethod proxyM
     public static boolean isHttpMethodSupportRepeatableRequestHeaders(HttpMethod httpMethod) {
         return REPEATABILITY_REQUEST_HTTP_METHODS.contains(httpMethod);
     }
+
+    /**
+     * @param requests a list of requests
+     * @return the first request consumes binary type, if no binary request, return the first request in requests
+     *
+     * If the binary request does not have content-type parameter, we will add one for it
+     */
+    public static Request findFirstBinaryRequest(List<Request> requests, Operation operation) {
+        Request selectedRequest = requests.get(0);
+        for (Request request : requests) {
+            if (request.getProtocol().getHttp().getKnownMediaType() != null
+                    && request.getProtocol().getHttp().getKnownMediaType().equals(KnownMediaType.BINARY)) {
+                // add contentType parameter
+                if (haveDifferentContentTypes(requests) && !hasContentTypeParameter(request)) {
+                    Parameter contentTypeParameter = createContentTypeParameter(request, operation);
+                    request.getParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getParameters()), contentTypeParameter);
+                    if (contentTypeParameter.isRequired()) {
+                        request.getSignatureParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getSignatureParameters()), contentTypeParameter);
```

Comment:
Since we need to pass in a list of requests to check if the requests have difference content types, will it be a bit odd to first find the binary request, then pass in a list of requests and add the content type parameter?

Reply:
Maybe call it `getSingleBinaryRequests`?